styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -204,8 +214,7 @@ private void processInstances(StyxConfig config, Map<String, Resource> resources
     var resourceExhaustedCache = new ConcurrentHashMap<String, Boolean>();
 
     // Shuffle the instances in order to process them in random order and reduce contention with other schedulers etc
-    var shuffledInstances = new ArrayList<>(activeInstances);
-    Collections.shuffle(shuffledInstances);
+    var shuffledInstances = shuffleInstances(activeInstances);
```

Comment:
this shuffle seems unnecessary now with the shuffle in the shuffleInstances method